### ISSUE TITLE
Multi-level delegation

### ISSUE THREAD

**Issue by kevgo** [Originate/rails-delegate-js#3]
As a developer building complex object nesting structures
I want to be able to delegate over multiple levels of nesting
So that I can expose deeply nested object hierarchies without unnecessary boilerplate.

**Comment by mcharawi** [comment]
I looked through the feature tests as well as trying it in my code, and it appears you also cannot delegate methods to multiple "subclasses", i.e.

```livescript
delegate \close, from:@, to: [@zmq-listener, @message-sender]
```
Maybe I missed something...

**Comment by kevgo** [comment]
That's correct. How would you see that work? If I call `superclass.close!`, should it call `superclass-zmq-listener.close!` or `message-sender.close!`? Or both?

**Comment by mcharawi** [comment]
I think that calling both would make the most sense, the goal here is to distribute the call to multiple subclasses so that you can avoid things like 
```livescript
delegate \close-listener, from: @, to: @zmq-listener
delegate \close-sender, from: @, to: @message-sender
```
which is repetitive and confusing. Particularly in this case, you would never call `close` on one and not the other.

**Comment by kevgo** [comment]
Okay, that makes sense. The code it replaces would be more like this though:

```ls
delegate \close, from: @, to: @zmq-listener
delegate \close, from: @, to: @message-sender
```

Or am I misunderstanding you?

Its a different ticket though. Can you please open a new ticket for this?

**Comment by mcharawi** [comment]
Well, that code wouldn't work because the `close` can only be delegated once per function, i.e. if your example runs, then it will delegate close to `@zmq-listener`, then un-delegate it and redelegate it to `@message-sender`.

Sure.

**Comment by kevgo** [comment]
Yeah, we might have to extend how this library works to make it work.
